File: go/currency/pkg/franc/franc.go
package franc

import m "github.com/chudee/currency-go-example/go/currency/pkg/money"

type Franc struct {
	m.Money
}

func NewFranc(amount int) *Franc {
	return &Franc{m.Money{amount}}
}

func (f *Franc) Times(multiplier int) *Franc {
	return NewFranc(f.Amount * multiplier)
}

File: ts/currency/src/franc.ts
export default class Franc {
  private amount: number;

  constructor(amount: number) {
    this.amount = amount;
  }

  times(multiplier: number): Franc {
    return new Franc( this.amount * multiplier)
  }

  equals(franc: Franc): boolean {
    return this.amount === franc.amount;
  }
}